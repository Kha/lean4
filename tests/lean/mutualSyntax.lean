mutual
notation "§" => Foo
inductive Foo where
  | mk : §
end
#print Foo
#check §
set_option pp.notation false in
#check §

mutual
local notation (priority := high) "§" => Foo.Bar
inductive Foo.Bar where
  | mk : §
end
#print Foo.Bar
set_option pp.notation false in
#check §

namespace Foo
mutual
notation "§§" => Baz
inductive Baz where
  | mk : §§
end
#print Foo.Baz
set_option pp.notation false in
#check §§
end Foo

mutual
notation:60 "⟨" c ", " σ "⟩"  " ⇓ " σ' => Bigstep c σ σ'
inductive Bigstep : Com → State → State → Prop where
  | skip : ⟨skip, σ⟩ ⇓ σ
end
#check ⟨0, 0⟩ ⇓ 0
