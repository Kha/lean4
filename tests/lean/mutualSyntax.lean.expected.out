inductive Foo : Type
number of parameters: 0
constructors:
Foo.mk : §
§ : Type
Foo : Type
inductive Foo.Bar : Type
number of parameters: 0
constructors:
Foo.Bar.mk : §
Foo.Bar : Type
inductive Foo.Baz : Type
number of parameters: 0
constructors:
Foo.Baz.mk : §§
Baz : Type
⟨0, 0⟩ ⇓ 0 : Prop
